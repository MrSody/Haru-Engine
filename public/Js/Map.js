class Map {
    constructor (data) {
        // sprite map
        this.spritesheet = new Image();
        this.spritesheet.src = data.spritesheet;
        // datos del sprite map
        this.spriteCol = 8;
        //spriteCol = parseInt(data.col);
        // tama√±o del map y sprite
        this.tileSize = parseInt(data.tileSize);
        this.capaOne;
        this.capaTwo;
        this.capaThree;
        this.capaFour;
        this.capaFive;
    }

/* ------------------------------ *
    GETTERS
 * ------------------------------ */

    getSpritesheet () {
        return this.spritesheet;
    }

    getTileSize () {
        return this.tileSize;
    }

/* ------------------------------ *
    SETTERS
 * ------------------------------ */

    setCapas (data) {
        this.capaOne = data.capa1;
        this.capaTwo = data.capa2;
        this.capaThree = data.capa3;
        this.capaFour = data.capa4;
        this.capaFive = data.capa5;
    }

/* ------------------------------ *
    FUNCIONES
 * ------------------------------ */

    // Dibuja Map - 100%
    drawMap (capa, ctx, w, h) {
        // Draw World - Dibujando Mundo
        let spriteNum = 0, x, y;

        spriteNum = capa[h][w];
        if (spriteNum != 0 && spriteNum != undefined) {

            // Trae la posicion del sprite
            if (spriteNum < this.spriteCol) {
                y = 0;
                x = parseInt((spriteNum - 1) * 32);
            } else {

                let ind = 0, fila = 0;

                while (!(ind > spriteNum)) {
                    fila++;
                    ind = (fila + 1) * 8
                }

                spriteNum = spriteNum - (fila * 8);
                y = parseInt(fila * 32);
                x = parseInt((spriteNum - 1) * 32);
            }
            
            // Mejora la posicion del mapa
            h = h - 0.5;

            ctx.drawImage(this.spritesheet, x, y, this.tileSize, this.tileSize, w * this.tileSize, h * this.tileSize, this.tileSize, this.tileSize);
            //ctx.drawImage(this.spritesheet, x, y, this.tileSize, this.tileSize, w * this.tileSize, h * this.tileSize, this.tileSize, this.tileSize);
        }
    }

    drawMapDown (ctx, w, h) {
        this.drawMap(this.capaOne, ctx, w, h);
        this.drawMap(this.capaTwo, ctx, w, h);
        this.drawMap(this.capaThree, ctx, w, h);
    }

    drawMapUp (ctx, w, h) {
        this.drawMap(this.capaFour, ctx, w, h);
        this.drawMap(this.capaFive, ctx, w, h);
    }
}