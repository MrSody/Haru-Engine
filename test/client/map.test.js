/**
 * @jest-environment jsdom
 */

const WORLD = require('../../engine/modules/world').World;

import Map from '../../public/game/js/modules/world/map.js';

const World = new WORLD();
const clsMap = new Map({spritesheet: World.dataSpriteSheets, tileSize: World.tileSize});

test('desingSheet', () =>{
    let response = [{
            name: "tileset_konoha_naruto_by_zewiskaaz-d55puro",
            spritesheet: '<img src="tileset_konoha_naruto_by_zewiskaaz-d55puro"/>',
            tileStart: 1,
            tileEnd: 816,
            tileCount: 816,
            imageHeight: 3264,
            imageWidth: 256,
        }];

    //expect(clsMap.desingSheet(World.getDataSpriteSheets())).toStrictEqual(response);
});

// TODO: Revisar - posSprite - 
test('posSprite', () => {
    let spriteNum = 9;
    let imageWidth = 256;
    let imageHeight = 3264;
    let response = {X: 0, Y: 1};

    //expect(clsMap.posSprite(spriteNum, imageWidth, imageHeight)).toStrictEqual(response);
});
