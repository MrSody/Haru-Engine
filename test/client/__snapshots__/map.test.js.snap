// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawMap 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAiAAAAFgCAYAAACCOcLtAAAABmJLR0QA/wD/AP+gvaeTAAAFRElEQVR4nO3dv24TQRAH4MmJwmXcWLbp3ERy4SbigRBvATW8BeIp6OkRDUUkGhqEbbkxpbtQRHOMN5eEOvd9Tfxn99L+NDO7vnj78dVtRMR0Nomb7z9jMV/Gbr+NiIjFfBnVdDaJ4+EUERFX15fx49ufiIjY7bex3qz671r5vLpmt9/GYr6MD2++XgxuAgCerS7DwfFw6gPHYr7sg8huv43pbNKvyfVfPt/EdDbpg8Rj4WO9WQ2uyecCAOPStUEjA0bEvyByPJzOwkKGjrZiMhQontoPAIxPl2GgrVDUILLbb8/Cwnqz6kNExL9WTVvhyGc8tR8AGJe+AnJ1fdl/mG2TOgtSw0ZWPobknqvry3stnfz+sf0AwPPXZdj48e1PtNWQGhoi4mzmo77Pv/ndbr89G1BN2YbJNWZAAGCcXkTcb508pIaSiOjbKu3nNbi062tV5H//LwDwvHTrzSoi7qoTNTjUgdQ6aFot5suo+9tqRw0fbRi5ur4cfCYA8Px1WYWoR2yHqhNtCMnA8VAVJD9vWzPp8OtkDgQARupFvVysDQSPVTCOh1Nf/ciwMdSGydd1/10oiX4/ADAuXVY+hoZC6/uhasXxcDqrlNQ2TD5vKHwMHfsFAMbj4vX7l7ftNepttSOrJG0rpZ6GeSxMPLQ/IuLTu9+uYgeAkenam0rbI7e7/fYsXOT9Hm07pu5vj+bWFs3QBWYAwLh0+SJDRs5l1LmQen9H3u/RBpa6v94jUk/K1PBh/gMAxquLOL8sLIPDdDa5N2RatYOrtVqSFZD2pEz93RnzHwAwXl0NH7X9UgdMa7WiVj7aG1DTerOKm+8/z/bnuqyQuAkVAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACe9heZWt1fkwWvqwAAAABJRU5ErkJggg=="`;

exports[`drawMapDown 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAiAAAAFgCAYAAACCOcLtAAAABmJLR0QA/wD/AP+gvaeTAAANa0lEQVR4nO3dS4wd6VkG4Leqzq0vbrttT4/dniTEAxpigiEXkGCFwkUgYMcCkRCBkk0EWxRBYMVVioRAbFCkAJEAISGkILFgN0isUBhFmUTDDJCZwMTudvvSdre7z6XqVLFwTqfHM2TpM9I8z6ZPdXdVHXWfxavv/+r7EwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA3kqx7DfwTvdbn/+hbmf3Zq5dv5qvffnrmc+7/Ns/jnPr1sOkSIqiSzXspU2RUX+Uw7v3c/HselaqIt1snPMXLufe/kHatk0xb3PmzJlM6zrNvMud+wf56G+/Pwfj20mSa9evZn9v8ob7/8EnvuQzAMAT11v2G3in29m9mcuXtvPSi6/mypUrSZIrn2rTtcl0PM8X/+zlDMph2q7JfDzOufW1nF9bz7AqM17pp2tnObO+kvHxNKurVebzaQb9MnUzTV13ufdgL9c//Gz29ybZ35uc3G8RegBgGcplv4F3usuXtk++bm6NkiRVr0xvUGa0WmXj4ih13aRs61y+eDYXN1czGhSZ1kdZG/ZSdE3S1VlfKdLvdzm73s/G2iDnN9cyGPay9dSl7O9NTq69CB+XL22/qRoCAE+KALJkO7s3T14/Hgj27u7mp375vWmacS5tbeb8+iBrK2U2zg6zcXYlZdFmWLVZHZRZHZZZHSbDfptmepCuHqdtmxRF8aZrX7t+9Q33BYAnzRLMkp2uSCxCwberIdu58c0bOTfqZ6Oosl4lRVklXZ3V9WGKci3jusm8TW7v7eaZS0+lqqqc2VjLaGUtd46+kXbe5vGc+dKLr55UXgBgGQSQt4FFc+jpULCoWFw4+3Sa4k4yKLN+7kyKtGnaeY6Pj7LSr7I+GKbtitTTtayuDVMVZW7u7GU6nebg7jR7t2+lf1ichJrT91ksywDAk2YJ5m3g8aWXRSVkZ/dm2nmXg3GTWZXcOtjPg+lxuqpInSZl1SVtk9n0KKPRIPNmlrNn1tIfVBn0R/mJj/xAnv+L/WxdvJyd3Zt56cVXv+N9AeBJEUCWbHNrlJ3dm2+qRmxujTI7avP8F/4rH772bMZ37mcwL7NRDTO+ez9PraylX1Y5Pj5K13VpZtMURZG6rlMVvdR1nYP93QyrJm3TJsmbll30gQCwLALIki36MU5XIy5f2s7ua0f50t/fy/T2Woa9WbpqIw9ns/QHZYoMsnFpmHNnqxTz1RS9LkXvTKbHw9w/2MnamWEmsyqrw16OD7o0szaXn37UY7IIHZZfAFgmAWTJFrM4nvvQuZPvffP1m/mHP/lqVtf7WVlNjsb3k6LJyspKjo+m6fdWc2vnMFWKjFaLTCd1xuOjDAa99KpRurLL4fFRjg+P85Gf/J588U9fyev/eyOXL22fNLvu703MAQFgaQSQJVv0Zbzywv1sbo1yc+dm/uUL97O+upa0w4zH42xsbGTeNpnNJulVZTbOrCZtlbZNymKc4WA9q6ujHE8fZjxtc+HiZpp6nkG1mgd791I0/Tz/V/czOZon+fZSzOM9IQDwpAggbwOL5ZD9vUm2L2/n4tP9PDw8zmSc1E2Vepq08y7DUZUuk4xW69TNw5RlmcGwS113qedter2kLPqpqkf/1i5Nhr2zaY5mWS17+bvPvpy26U6qLR7FBWBZBJAlW/R/nO7J+PGPP5tf+b0P5gM/t5KyaDKd1hmNRmnqNr1eP71+l3ZepGurTJtZuhQp0suknmQ2m6aumwx6RdreJEdHR/nAh55JUx9mfVTkzz/z1bzy7/eTaEIFYHkEkCVbNIYu9mnZ3BqlPyzz8MEsX/6nO3nu2tNZWSszm81SFqPMZmVSDNJ1Z3Lr9r30B2uZp8v+g3FWVlYyGJa5deN2rlx5KnfutJm3yfHR3UzHZfbv1zn3dJmmabO5NVIBAWBpDCJbssdDwOJpmM2nRmlmTXpVnXo6S1kM0qZKWfazu7ebNmuper3cuXc/02aQpuml64pcuLievd1JymKa43ol9Xyc1dWV9PuT/MJvPJf+qMrtu7vp9bdVQABYGhWQJXurEHDuqVFe+sprmRy3mc+6NHWRuu5y+/btdOU8VX+Wrhyn7ZLJtM1gMEhVDjKZzNKvkvm0S5l52mqaw8lh5k0/7/qujfzNZ1/Oxe2R3XABWDoBZMneaiz6f3zl1fzrX9/JDz/37tTzBzmu21RlL0WznsOjw9RNL2v9YaZNnX61npvfvJeuOM60TvbuJU2vzrROhl1ycL9NXVbfOq6z9/rxyX1MQgVgWQSQJVtUQE7v0XJhcyv1uMvDyTRlNchgZTV12+R4fJCtrYvZ2NhI2z7q75jM6lRVP22bdF2R6XSaousyn9UZrozSpsvu7l6OJ20uPn0hf/vH/3lyb8PIAFgWPSBLdnoX3GvXr+ZrL3w9//y5/bzn3U9nUteZt7Mcjg9TtWeTosm0qTPoFannTe4f1pk1VeZdlaaZp553adukKvspiiJNV6dLmUmdzO4dZzaZZlaWuXHjRq5cuWIOCABLI4As2c7uzZPKx/7eJEVVpGjmmUybFOUgD8eTpFzLw8OjrA2q9Kph7u7fyXhc5mgyz7RJXr9xOxfOn82gN836Si9dO0/btpk1TZo2mXXjtHUyXOnn/LlB3v+BZ09GwCc3lvsHAOAdSQBZskUF5Md+9lpeeeHRfI7+mSI3dvaTtklRlqnKKt2szda7z+e1b+ymrut03Sh128/d/YfprYxyeNxk0C/z4ME4F85v5ODgIEWqNG3yid+9nqoqMhhVubFz81T4AIDlEECWbPE0yisv3M/O7s1sb2/no5++nNl4nnbepSiK7OzcyvOfv5fJZJbV1UHqNqnrMg+OxtmfTvNrf/TBfO53vpybu0eZnV/J0XiW4XCYppmn67oMV6uUZfFo591b8RQMAEunCfVt4PGnUcqqyGi9l+FqL6P1Kv1RUvXblP0y+w8OUrdVHhyNc/v+QX7mkxfTFV0+9unvy5mttXTlIPcOxjmePpqUmrJM13RJ8obKx+M78ALAkySALNmiCnF6MuliIure3Z0UZZFn3rWd2WyeLvWj5tJ2nvFkkmptmB/8ke9Okmw/eyY/+osbOZrV6YpBHo6nefjwKEW/TNkv39BrkjzafdcgMgCWRQBZskUVYjGKPXlzdaLrkqKoMpn2c3A0yM6t49SjKj/9yc28/LXXTs5/z3ufyS995n25c1xnf9xk1g4znTbZ/Vb4OB049l6f6AMBYGkEkCU7PYvj8UCwOC6rImW/y8OHk9Rtl2q9yMc+/b25/uFnc+361ZP9ZJKkNyjzqT/8/qRqc+/oYTYuVNm6eOkNFZCd3UeNqOaAALAsmlCX7HTl49r1q2+ofGxujU6Or16/kP/578O0mefjv/m+FGXxph6Oza3RSZ/Hz//6+ZPrPR4+Tj/2CwDLUCz7DbzT/ervX+keDx6P92ssgsjOY0spp8e4f6cw8f+dnyR/+ZkbPgMAPHGWYJZs0e+xWA5ZhIPNrdHJ0srpcPHch87l8qXtNwSUx89fNLEuXi8mni7Cx+knYQBgGQSQt4lFyFg8FbM4vnxp+yRcXL60fTKs7PHAcvr8RahZBJXFNU+HDzNAAFgmAeRt4PSSyCI4bG6NTkLCW+3ZcjqgLK6xqJYsKiCLKsjjFZDHqyoA8KQJIEt2OnycXn7Z35u8qSqy+Nnid08vs5xeTrl2/WpeevHVN5y/+L1FhWRxPgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAbw//B8e9HEUaO9bYAAAAAElFTkSuQmCC"`;

exports[`drawMapLayers 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAiAAAAFgCAYAAACCOcLtAAAABmJLR0QA/wD/AP+gvaeTAAANa0lEQVR4nO3dS4wd6VkG4Leqzq0vbrttT4/dniTEAxpigiEXkGCFwkUgYMcCkRCBkk0EWxRBYMVVioRAbFCkAJEAISGkILFgN0isUBhFmUTDDJCZwMTudvvSdre7z6XqVLFwTqfHM2TpM9I8z6ZPdXdVHXWfxavv/+r7EwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA3kqx7DfwTvdbn/+hbmf3Zq5dv5qvffnrmc+7/Ns/jnPr1sOkSIqiSzXspU2RUX+Uw7v3c/HselaqIt1snPMXLufe/kHatk0xb3PmzJlM6zrNvMud+wf56G+/Pwfj20mSa9evZn9v8ob7/8EnvuQzAMAT11v2G3in29m9mcuXtvPSi6/mypUrSZIrn2rTtcl0PM8X/+zlDMph2q7JfDzOufW1nF9bz7AqM17pp2tnObO+kvHxNKurVebzaQb9MnUzTV13ufdgL9c//Gz29ybZ35uc3G8RegBgGcplv4F3usuXtk++bm6NkiRVr0xvUGa0WmXj4ih13aRs61y+eDYXN1czGhSZ1kdZG/ZSdE3S1VlfKdLvdzm73s/G2iDnN9cyGPay9dSl7O9NTq69CB+XL22/qRoCAE+KALJkO7s3T14/Hgj27u7mp375vWmacS5tbeb8+iBrK2U2zg6zcXYlZdFmWLVZHZRZHZZZHSbDfptmepCuHqdtmxRF8aZrX7t+9Q33BYAnzRLMkp2uSCxCwberIdu58c0bOTfqZ6Oosl4lRVklXZ3V9WGKci3jusm8TW7v7eaZS0+lqqqc2VjLaGUtd46+kXbe5vGc+dKLr55UXgBgGQSQt4FFc+jpULCoWFw4+3Sa4k4yKLN+7kyKtGnaeY6Pj7LSr7I+GKbtitTTtayuDVMVZW7u7GU6nebg7jR7t2+lf1ichJrT91ksywDAk2YJ5m3g8aWXRSVkZ/dm2nmXg3GTWZXcOtjPg+lxuqpInSZl1SVtk9n0KKPRIPNmlrNn1tIfVBn0R/mJj/xAnv+L/WxdvJyd3Zt56cVXv+N9AeBJEUCWbHNrlJ3dm2+qRmxujTI7avP8F/4rH772bMZ37mcwL7NRDTO+ez9PraylX1Y5Pj5K13VpZtMURZG6rlMVvdR1nYP93QyrJm3TJsmbll30gQCwLALIki36MU5XIy5f2s7ua0f50t/fy/T2Woa9WbpqIw9ns/QHZYoMsnFpmHNnqxTz1RS9LkXvTKbHw9w/2MnamWEmsyqrw16OD7o0szaXn37UY7IIHZZfAFgmAWTJFrM4nvvQuZPvffP1m/mHP/lqVtf7WVlNjsb3k6LJyspKjo+m6fdWc2vnMFWKjFaLTCd1xuOjDAa99KpRurLL4fFRjg+P85Gf/J588U9fyev/eyOXL22fNLvu703MAQFgaQSQJVv0Zbzywv1sbo1yc+dm/uUL97O+upa0w4zH42xsbGTeNpnNJulVZTbOrCZtlbZNymKc4WA9q6ujHE8fZjxtc+HiZpp6nkG1mgd791I0/Tz/V/czOZon+fZSzOM9IQDwpAggbwOL5ZD9vUm2L2/n4tP9PDw8zmSc1E2Vepq08y7DUZUuk4xW69TNw5RlmcGwS113qedter2kLPqpqkf/1i5Nhr2zaY5mWS17+bvPvpy26U6qLR7FBWBZBJAlW/R/nO7J+PGPP5tf+b0P5gM/t5KyaDKd1hmNRmnqNr1eP71+l3ZepGurTJtZuhQp0suknmQ2m6aumwx6RdreJEdHR/nAh55JUx9mfVTkzz/z1bzy7/eTaEIFYHkEkCVbNIYu9mnZ3BqlPyzz8MEsX/6nO3nu2tNZWSszm81SFqPMZmVSDNJ1Z3Lr9r30B2uZp8v+g3FWVlYyGJa5deN2rlx5KnfutJm3yfHR3UzHZfbv1zn3dJmmabO5NVIBAWBpDCJbssdDwOJpmM2nRmlmTXpVnXo6S1kM0qZKWfazu7ebNmuper3cuXc/02aQpuml64pcuLievd1JymKa43ol9Xyc1dWV9PuT/MJvPJf+qMrtu7vp9bdVQABYGhWQJXurEHDuqVFe+sprmRy3mc+6NHWRuu5y+/btdOU8VX+Wrhyn7ZLJtM1gMEhVDjKZzNKvkvm0S5l52mqaw8lh5k0/7/qujfzNZ1/Oxe2R3XABWDoBZMneaiz6f3zl1fzrX9/JDz/37tTzBzmu21RlL0WznsOjw9RNL2v9YaZNnX61npvfvJeuOM60TvbuJU2vzrROhl1ycL9NXVbfOq6z9/rxyX1MQgVgWQSQJVtUQE7v0XJhcyv1uMvDyTRlNchgZTV12+R4fJCtrYvZ2NhI2z7q75jM6lRVP22bdF2R6XSaousyn9UZrozSpsvu7l6OJ20uPn0hf/vH/3lyb8PIAFgWPSBLdnoX3GvXr+ZrL3w9//y5/bzn3U9nUteZt7Mcjg9TtWeTosm0qTPoFannTe4f1pk1VeZdlaaZp553adukKvspiiJNV6dLmUmdzO4dZzaZZlaWuXHjRq5cuWIOCABLI4As2c7uzZPKx/7eJEVVpGjmmUybFOUgD8eTpFzLw8OjrA2q9Kph7u7fyXhc5mgyz7RJXr9xOxfOn82gN836Si9dO0/btpk1TZo2mXXjtHUyXOnn/LlB3v+BZ09GwCc3lvsHAOAdSQBZskUF5Md+9lpeeeHRfI7+mSI3dvaTtklRlqnKKt2szda7z+e1b+ymrut03Sh128/d/YfprYxyeNxk0C/z4ME4F85v5ODgIEWqNG3yid+9nqoqMhhVubFz81T4AIDlEECWbPE0yisv3M/O7s1sb2/no5++nNl4nnbepSiK7OzcyvOfv5fJZJbV1UHqNqnrMg+OxtmfTvNrf/TBfO53vpybu0eZnV/J0XiW4XCYppmn67oMV6uUZfFo591b8RQMAEunCfVt4PGnUcqqyGi9l+FqL6P1Kv1RUvXblP0y+w8OUrdVHhyNc/v+QX7mkxfTFV0+9unvy5mttXTlIPcOxjmePpqUmrJM13RJ8obKx+M78ALAkySALNmiCnF6MuliIure3Z0UZZFn3rWd2WyeLvWj5tJ2nvFkkmptmB/8ke9Okmw/eyY/+osbOZrV6YpBHo6nefjwKEW/TNkv39BrkjzafdcgMgCWRQBZskUVYjGKPXlzdaLrkqKoMpn2c3A0yM6t49SjKj/9yc28/LXXTs5/z3ufyS995n25c1xnf9xk1g4znTbZ/Vb4OB049l6f6AMBYGkEkCU7PYvj8UCwOC6rImW/y8OHk9Rtl2q9yMc+/b25/uFnc+361ZP9ZJKkNyjzqT/8/qRqc+/oYTYuVNm6eOkNFZCd3UeNqOaAALAsmlCX7HTl49r1q2+ofGxujU6Or16/kP/578O0mefjv/m+FGXxph6Oza3RSZ/Hz//6+ZPrPR4+Tj/2CwDLUCz7DbzT/ervX+keDx6P92ssgsjOY0spp8e4f6cw8f+dnyR/+ZkbPgMAPHGWYJZs0e+xWA5ZhIPNrdHJ0srpcPHch87l8qXtNwSUx89fNLEuXi8mni7Cx+knYQBgGQSQt4lFyFg8FbM4vnxp+yRcXL60fTKs7PHAcvr8RahZBJXFNU+HDzNAAFgmAeRt4PSSyCI4bG6NTkLCW+3ZcjqgLK6xqJYsKiCLKsjjFZDHqyoA8KQJIEt2OnycXn7Z35u8qSqy+Nnid08vs5xeTrl2/WpeevHVN5y/+L1FhWRxPgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAbw//B8e9HEUaO9bYAAAAAElFTkSuQmCC"`;

exports[`drawMapUp 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAiAAAAFgCAYAAACCOcLtAAAABmJLR0QA/wD/AP+gvaeTAAAFk0lEQVR4nO3cQW4bVRwG8K+4JG2ckrSk2BVq2FBFXbe73oEbwIYNa5ZcgQ07TsAJugBxAM6AgA0tUmMZySmxnTptVRbRTNMQj9NKyeiZ32/jkf6e0VtZn755zwkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMddansBQLMvP9t59dXnXyRJBk8ep3frdv35zbffJ0nu391Jv7eZl4eHGY6m6fc2M54+z8e9Xn54+GPu393J+rVurl5+9cazv/7uod8AoBXvtb0A4GxOho/Bk8f17Ob1tfp6/Vr36HPt/cbnHbyQPYD2CCBQiJPho3frdj0bjqZJks7KSsb7k6yurmQ8fd74vPH+5FzXC9BEAIFCzGtAtvtHjcfuYC9JsvpBN7PZYX1fNa8akeFoms7KSt2UALRBAIFCzGtAHu1OjvZ8PJslSWb/vG42tjY6ebQ7eadXNADnSQCBQjTtAUmS9SurdQtS+fvpyyTv9ooG4DwJIFCIeQ3Idr+b3cFexs9mb7xWGe9PsrXRyXa/+58GZDY71IAArRJAoBBNDUi/t5n1K6v1xtLRwVG7UTUgx1Xf0YAAbRJAoBBNe0BONiDVPpBqD8hwNM0ffw7q0LE72HMKBmiVAAKFeJsGpAoiVQMybw7QFgEECtH0PyBJ6lMwyevXLFsbncY5QFsEECjEWU7BVCHjtIZj0RzgIgkgUIizNCCfftI7uj6l4Vg0B7hIAggUYlEDMj0Y1tenNRyL5gAXSQCBQsxrQCaj9frYbeV4w7FoDtAGAQQKMa8B6V4fJ0nWrt489b5Fc4A2CCBQiKYGJEk+/OhGkuTBvTvp9zbz4N6d/DV82jgHaMvlthcAnE3THpCff/qt8d5Fc4CLJoBAIeY1IL/8/uullpcG8Na8goFCLDoFA1ASAQQKseh/QABKIoBAITQgwDIRQKAQGhBgmQggUAgNCLBMBBAohAYEWCYCCBRCAwIsEwEECqEBAZaJAAKF0IAAy0QAgUJoQIBlIoBAITQgwDIRQKAQGhBgmQggUAgNCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD8P/wL0cW98MDfULcAAAAASUVORK5CYII="`;
